<?xml version="1.0"?>
<launch>
    <arg name="global_frame_id" default="map" />
    <!-- gmapping parameters -->
    <arg name="delta" default="0.05"/>

    <arg name="minimumScore" default="550.0"/>
    <arg name="maxUrange" default="20.0"/>
    <arg name="srr" default="0.05"/>
    <arg name="srt" default="0.05"/>
    <arg name="str" default="0.05"/>
    <arg name="stt" default="0.05"/>
    
    <arg name="odom_frame" default="slam"/>
    <arg name="map_frame" default="map" />

    <arg name="xmin" default="-100"/>
    <arg name="ymin" default="-100"/>
    <arg name="xmax" default="100"/>
    <arg name="ymax" default="100"/>
    
    <!--<arg name="map_folder" default="/storage/predicted_maps/" />-->
    
    <arg name="machine" default="localhost" />
    <arg name="user" default="" />

    <!-- NOW when launching in a remote mode it will need the ROS_ENV_LOADER set if not it will leave it empty -->
    <machine name="$(arg machine)" address="$(arg machine)" env-loader="$(optenv ROS_ENV_LOADER )" user="$(arg user)" default="true" />
    

   
    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" respawn="true">
	<param name="xmin" value="$(arg xmin)"/>
	<param name="ymin" value="$(arg ymin)"/>
	<param name="xmax" value="$(arg xmax)"/>
	<param name="ymax" value="$(arg ymax)"/>
	<param name="delta" value="$(arg delta)"/>
	<param name="maxUrange" value="$(arg maxUrange)"/>

	<param name="srr" value="$(arg srr)"/>
	<param name="srt" value="$(arg srt)"/>	
	<param name="str" value="$(arg str)"/>
	<param name="stt" value="$(arg stt)"/>
	<param name="minimumScore" value="$(arg minimumScore)"/>
	
	<param name="odom_frame" value="$(arg odom_frame)"/>
	<param name="map_frame" value="$(arg map_frame)" />
	<remap from="/map" to="/gmap"/>

    </node>
    
    <!--<node pkg="aaf_bringup" type="save_and_upload_maps.py" name="save_and_upload_maps" respawn="true">
      <param name="map_folder" value="$(arg map_folder)"/>
    </node>-->
</launch>
